OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```
@@ -374,140 +315,144 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
+                                    val ot = p.toString.replaceAll(otRegex, "")
+                                    //only interfaces defined in 'callbacklist' are handled as callback interfaces and dynamically searching for callback interfaces
+                                    if (callbackList.contains(ot) ||
+                                        (ot.startsWith("android") && ot.endsWith("Listener") && ot.substring(ot.lastIndexOf("$") + 1).startsWith("On"))) {
+                                        val par = methodCall.asVirtualMethodCall.params(i).asVar.value.verificationTypeInfo
+                                        if (par.isObjectVariableInfo) {
+                                            val cf = project.classFile(par.asObjectVariableInfo.clazz.asObjectType)
+                                            if (cf.isDefined) {
+                                                val methods = cf.get.methods.filter(f ⇒ f.name != init && f.name != "<clinit>")
+                                                methods.foreach { method ⇒
+                                                    if (!callbackMethods.contains(method)) callbackMethods += method
                                                 }
                                             }
-                                        } else i += 1
+                                        }
                                     } else i += 1
-                                }
+                                } else i += 1
                             }
                         }
-                }
-                if (callbackMethods.nonEmpty) {
-                    generateCallBackEdges(m, callbackMethods)
-                }
-            } catch {
-                case _: Throwable ⇒
+                    }
             }
-        })
-
+            if (callbackMethods.nonEmpty) {
+                generateCallBackEdges(m, callbackMethods)
+            }
+        } catch {
```

This empty catch seems like a really bad idea. What exceptions do you expect here? Is it really ok to not handle them at all? Should there at least be a warning of some kind?

---

It was meant to catch methods, where no tac representation  was available. It is no longer necessary, as the tac representation is provided by the property store.